src/main/java/com/ryszardpanda/medicalClinic/PatientController.java

```diff
@@ -0,0 +1,55 @@
+package com.ryszardpanda.medicalClinic;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController // mowie ze to jest Bean Springowy oraz kontroler -> czyli tutaj bede przechwytywal zapytania
+@RequiredArgsConstructor
+public class PatientController {
+    private final PatientService patientService;
+
+    // to jest po prostu definicja endpointu na sciezce "/patients"
+    // GET /patients
+    @GetMapping("/patients")
+    public List<Patient> getPatients() {
+        return patientService.getPatients();
+    }
+
+    @PostMapping("/patients")
+    public List<Patient> addPatients(@RequestBody List<Patient> patients) {
+        return patientService.addPatients(patients);
+    }
+
+    @GetMapping("/patients/{email}")
+    public Patient getPatientByEmail(@PathVariable String email) {
+        return patientService.getPatientByEmail(email);
+    }
+
+    // Metoda DELETE, która usunie pacjenta na podstawie adresu e-mail
+    //nie ogarnalbym
+    @DeleteMapping("/patients/{email}")
+    public ResponseEntity<Void> deletePatientByEmail(@PathVariable String email) {
```

Comment:
ResponseEntity to klasa reprezentująca odpowiedź HTTP. W sytuacji jak używasz adnotacji @RestController to automatycznie masz zamiane tego co zwracasz czyli np tak jak w powyzszym endpoincie zwracasz Patient to ten Patient trafia do body a Spring automatycznie obudowuje to body (Patient) w odpowiedź HTTP z statusem sukcesu 200 OK jeśli zapytanie się powiedzie. Defaultowy mechanizm Springa w sytuacji kiedy poleci w ramach przetwarzania zapytania wyjątek automatycznie zmapuje to na odpowiedź HTTP z kodem 500 INTERNAL_SERVER_ERROR.